### DISCUSSION TITLE
How MTEX calculates unique variants of an Orientation Relationship

### DISCUSSION THREAD

**Discussion by lebraunjames** [mtex-toolbox/mtex#1876] [Ask Anything]
Hello MTEX Community,

I'm trying to understand how MTEX handles unique variant identification in Orientation Relationships (OR). Specifically, I'm working with the Shoji-Nishiyama OR:
(111)γ-fcc || (0001)ε-hcp
[1¯10]γ-fcc || [11¯20]ε-hcp

I define the OR with this syntax:
```matlab
mori = orientation.map(Miller(1, 1, 1,cs_fcc),Miller(0, 0, 0, 1,cs_hcp),...
      Miller(1, -1, 0,cs_fcc,'uvw'),Miller(1, 1, -2, 0,cs_hcp,'uvw'));
```
However, when I check the variants: 
```matlab
round2Miller(mori.variants(1)), round2Miller(mori.variants(2)), round2Miller(mori.variants(3)), round2Miller(mori.variants(4))
```
I get an unexpected hcp direction:
```
(111) || (0001)   [10-1] || **[2-1-10]**
(1-11) || (0001)   [10-1] || **[-2110]**
(1-1-1) || (000-1)   [01-1] || **[2-1-10]**
(11-1) || (0001)   [1-10] || **[2-1-10]** 
```
We've separately identified eight variants in R-F space with the correct directions. For instance, two of them look like this:
```
(1 1 1) // (0 0 0 1), [1 -1 0] // [-1 -1 2 0]
(1 -1 1) // (0 0 0 1), [-1 -1 0] // [-1 -1 2 0]
```
Is there a way to manually input the variants when defining the OR or rectify the variants identified by the OR in MTEX? Any clarification or guidance is much appreciated.

Best,
James

**Comment by ralfHielscher** [comment]
Could you please explain more explicitly what you think is not correct in MTEX.

Maybe one point is that applying a symmetry operation of the child phase does not change the variant.

Ralf.

**Reply by lebraunjames** [reply]
Hello Ralf,

Thank you for your response.

In utilizing the built-in Shoji-Nishiyama OR:
job.csParent = cs_fcc;
job.csChild = cs_hcp;
job.p2c = orientation.ShojiNishiyama(job.csParent, job.csChild)

And subsequently checking the variants, the derived directions do not align entirely with those reported in literature:
round2Miller(job.p2c.variants(1))...round2Miller(job.p2c.variants(4))

(111) || (0001)   [10-1] || [2-1-10]
(1-11) || (0001)   [10-1] || [-2110]
(1-1-1) || (000-1)   [01-1] || [2-1-10]
(11-1) || (0001)   [1-10] || [2-1-10]

I'm expecting [1-10] and [11-20] as per the literature, but I get [10-1] and [2-1-10] instead.
I attempted to input the relationship explicitly:

mori = orientation.map(Miller(1, 1, 1,cs_fcc),Miller(0, 0, 0, 1,cs_hcp),...
Miller(1, -1, 0,cs_fcc,'uvw'),Miller(1, 1, -2, 0,cs_hcp,'uvw'));

and checked the variants:
round2Miller(mori.variants(1))...round2Miller(mori.variants(4))
(111) || (0001)   [10-1] || [-12-10] 
(1-11) || (0001)   [110] || [2-1-10] 
(1-1-1) || (000-1)   [01-1] || [-12-10] 
(11-1) || (0001)   [1-10] || [-12-10]

However, I'm still encountering an issue with the hcp directions, with a result of [-12-10] instead of the input of [11-20].

Is there a way to correct this difference and align the variants with the directionalism condition?

Best regards,
James

**Comment by ralfHielscher** [comment]
The point is that two variants
```matlab
mori1 = orientation.ShojiNishiyama(cs_fcc, cs_hcp);
mori2 = orientation.map(Miller(1, 1, 1,cs_fcc),Miller(0, 0, 0, 1,cs_hcp),...
  Miller(1, -1, 0,cs_fcc,'uvw'),Miller(1, 1, -2, 0,cs_hcp,'uvw'));
```
are symmetrically equivalent, if there is a hcp symmetry `sym` such that `mori1 = sym * mori2`.
Indeed
```matlab
mori1
round2Miller(cs_hcp*mori2)
```
shows
```
mori1 = misorientation (432 → 622)
 
 (111) || (0001)   [10-1] || [2-1-10]

cs_hcp * mori2 = misorientation (432 → 622) 
  plane parallel   direction parallel    fit
(111) || (0001)    [10-1] || [-12-10]     0°
(111) || (000-1)   [01-1] || [-2110]     0°
(111) || (0001)    [1-10] || [-12-10]     0°
(111) || (000-1)   [01-1] || [-12-10]     0°
(111) || (0001)    [01-1] || [1-210]     0°
(111) || (000-1)   [10-1] || [-12-10]     0°
(111) || (0001)    [01-1] || [2-1-10]     0°
(111) || (000-1)   [01-1] || [2-1-10]     0°
(111) || (0001)    [10-1] || [2-1-10]     0°
(111) || (000-1)   [10-1] || [2-1-10]     0°
(111) || (0001)    [01-1] || [-12-10]     0°
(111) || (000-1)   [1-10] || [2-1-10]     0°
```
and hence `mori1` is identical to the fourth line from the bottom. Hence both the variants `mori1` and `mori2` are identical. 

I hope this helps,

Ralf.

**Reply by lebraunjames** [reply]
Hello Ralf,

Thank you for your previous detailed response.

However, I'm encountering a challenge when trying to replicate the symmetrically equivalent variants you showed. I'm using the following inputs:
```matlab
CS = {... 
  'notIndexed',...
  crystalSymmetry('6/mmm', [2.5 2.5 4.1], 'X||a*', 'Y||b', 'Z||c*', 'mineral', 'HCP Co', 'color', [0.53 0.81 0.98]),...
  crystalSymmetry('m-3m', [3.6 3.6 3.6], 'mineral', 'FCC Co', 'color', [0.56 0.74 0.56])};

cs_fcc = ebsd('FCC Co').CS;
cs_hcp = ebsd('HCP Co').CS;

mori1 = orientation.ShojiNishiyama(cs_fcc, cs_hcp);
mori2 = orientation.map(Miller(1, 1, 1,cs_fcc),Miller(0, 0, 0, 1,cs_hcp),...
  Miller(1, -1, 0,cs_fcc,'uvw'),Miller(1, 1, -2, 0,cs_hcp,'uvw'));

mori1
round2Miller(cs_hcp*mori2)
```
My results, however, are as follows:
```
plane parallel   direction parallel    fit
(111) || (0001)    [10-1] || [010]       0°
(111) || (000-1)   [01-1] || [-100]      0°
...
(111) || (0001)    [01-1] || [-100]     0°
(111) || (000-1)   [1-10] || [010]      0°
(111) || (0001)    [01-1] || [010]       0°
(111) || (000-1)   [01-1] || [0-10]     0°
(111) || (0001)    [10-1] || [010]       0°
```
Unfortunately, these results don't match the expected symmetrically equivalent variants you demonstrated. Could this discrepancy be a bug in MTEX, or might there be an error in my input parameters?

Your continued assistance is greatly appreciated.

Best regards,
James

**Reply by ralfHielscher** [reply]
The difference is because you are probably using MTEX 5.9 and I MTEX 5.10. However, this is only a display issue
If you check
```matlab
angle(mori1,cs_hcp*mori2,'noSymmetry')
```
You observe that mori1 matches one of the symmetrically equivalents of mori2.

Ralf.

**Reply by lebraunjames** [reply]
Thank you for your help in clarifying, Ralf.
